pilot/pkg/security/authn/v1beta1/policy_applier.go

```diff
@@ -281,3 +308,67 @@ func convertToEnvoyJwtConfig(jwtRules []*v1beta1.JWTRule) *envoy_jwt.JwtAuthenti
 		Providers: providers,
 	}
 }
+
+// GetMutualTLSMode retuns the MutualTLSMode enum corresponding to the given peer authentiation policy.
+// If the input authentication is nil, returns MTLSPermissive.
+func GetMutualTLSMode(peer *v1beta1.PeerAuthentication) model.MutualTLSMode {
+	if peer == nil || peer.Mtls == nil {
+		return model.MTLSPermissive
+	}
+	switch peer.Mtls.Mode {
+	case v1beta1.PeerAuthentication_MutualTLS_DISABLE:
+		return model.MTLSDisable
+	case v1beta1.PeerAuthentication_MutualTLS_PERMISSIVE:
+		return model.MTLSPermissive
+	case v1beta1.PeerAuthentication_MutualTLS_STRICT:
+		return model.MTLSStrict
+	default:
+		// Should never reach here
+		log.Errorf("Unknown peer authentication %#v", peer)
+		return model.MTLSUnknown
+	}
+}
+
+// GetMostSpecificScopeConfig returns the config with the most specific scope, i.e config with
+// non-emtpy selector should be preferred over those without, one in non-root namespace should be
+// preferred over those in root namespace. Where there are more than one config in the same
+// scope, pick the one with stronger security (STRICT > PERMISSIVE > DISABLE).
+// The input configs is assumed to be a short list of configs that should be considerred for an
+// workload.
+func GetMostSpecificScopeConfig(rootNamespace string, configs []*model.Config) *model.Config {
+	// 0 - workload-specific config, define in the same namespace as workload (non-root namespace).
+	// 1 - workload-specific config, define in the root namespace.
+	// 2 - namespace level config (i.e config without selector, in non-root namespace)
+	// 3 - mesh level config (i.e config without selector, in root-namespace)
+	configByScope := make([]*model.Config, 4)
+
+	for _, cfg := range configs {
+		spec := cfg.Spec.(*v1beta1.PeerAuthentication)
+		level := 0
+		if cfg.Namespace == rootNamespace {
+			level = 1
+		}
+		if spec.Selector == nil || len(spec.Selector.MatchLabels) == 0 {
+			level += 2
+		}
+		if isStronger(spec, configByScope[level]) {
+			configByScope[level] = cfg
+		}
+	}
+
+	for _, cfg := range configByScope {
+		if cfg != nil {
+			return cfg
+		}
+	}
+
+	return nil
+}
+
+func isStronger(spec *v1beta1.PeerAuthentication, cfg *model.Config) bool {
+	if cfg == nil {
+		return true
+	}
+
+	return GetMutualTLSMode(spec) > GetMutualTLSMode(cfg.Spec.(*v1beta1.PeerAuthentication))
```

Comment:
The order of the enum is more or less the same, and it's fairly intuitive, IMO. We can improve documentation to promote this behavior.

It could be fine to use the tie-breaker approach we used in other resources (sidecar, DR etc), e.g use the one with the latest timestamp. However, I feel this is not a good model for security.